clean-abap/sub-sections/InterfacesVsAbstractClasses.md

```diff
@@ -0,0 +1,186 @@
+# Interfaces vs. Abstract Classes
+
+Although interfaces and abstract classes share some properties,
+they are not equivalent and you should not treat one
+as an alternative for the other.
+
+Interfaces are there to share definitions.
+They specify how things are supposed to interact,
+without imposing any expectations
+what the implementation should look like.
+
+Abstract classes are there to share implementations.
+They also specify interaction,
+but in a way that already plots a path how to implement it,
+probably even assisting that implementation with code.
+
+Instead of asking _"Should I use an interface or an abstract class?"_,
+usually the clean way will be to state that _"I will use an interface, full stop"_,
+followed by the question _"Should I use an abstract class to implement it?".
+
+## Interfaces
+
+```ABAP
+INTERFACE /clean/blog_post.
+  PUBLIC.
+    METHODS publish.
+ENDINTERFACE.
+```
+
+Interfaces allow connecting any kind of code,
+as long as it satisfies the interface.
+
+They don't impose any
+side expectations or restrictions upon the implementation.
+
+In turn, they don't help implementing the required methods
+with default code or helper methods.
+
+```ABAP
+CLASS /clean/markdown_blog_post DEFINITION PUBLIC CREATE PUBLIC.
+  PUBLIC SECTION.
+    INTERFACES /clean/blog_post.
+ENDCLASS.
+
+CLASS /clean/markdown_blog_post IMPLEMENTATION.
+  
+  METHOD publish.
+  ENDMETHOD.
+  
+ENDCLASS.
+```
+
+![](interfaces-vs-abstract-classes/InterfacesVsAbstractClasses-Interface.png)
+
+> **Class diagram.**
+The `BlogPost` interface has two alternative
+implementations `MarkdownBlogPost` and `HTMLBlogPost`.
+Both fulfill the same "contract", specified by  `BlogPost`,
+but don't share any code.
+
+## Abstract classes
+
+```ABAP
+CLASS /clean/blog_post DEFINITION PUBLIC ABSTRACT CREATE PROTECTED.
+  PUBLIC SECTION.
+    METHODS publish ABSTRACT.
+ENDCLASS.
+
+CLASS /clean/blog_post IMPLEMENTATION.
+ENDCLASS.
+```
+
```

Comment:
```suggestion
An abstract class differs from regular classes in that it can’t be instantiated. Only 
sub-classes of the abstract class can be instantiated. An abstract class should contain 
at least one abstract method. An abstract method is declared in the abstract class, 
and then implemented in the sub-classes of the abstract class.

```

Reply:
I thought it would make sense to add a short intro to abstract classes and methods for readers less familiar with the concept.